samples/service.js

```diff
@@ -24,30 +34,60 @@ function createService (config) {
   }
 
   function readSample (id) {
-    const path = join(samplesDirectory, id)
+    const path = _getSamplePath(id)
 
     return fs.readFile(path)
-      .then(({ buffer }) => {
+      .then(data => {
+        const { buffer } = data
         return audioContext.decodeAudioData(buffer)
+          .then(audioBuffer => ({ data, audioBuffer }))
       })
-      .then(audioBuffer => ({ id, path, audioBuffer }))
+      .then(({ data, audioBuffer }) => ({ id, path, data, audioBuffer }))
   }
 
   function createSample (file) {
     const { path } = file
 
-    return fs.readFile(path)
-      .then(({ buffer }) => {
-        // TODO: dedupe against checksum
-        const id = checksum(arrayBufferToBuffer(buffer))
+    return fs.readFile(path).then((data) => {
+      const { buffer } = data
+      // TODO: dedupe against checksum
+      const id = checksum(arrayBufferToBuffer(buffer))
+      const targetPath = _getSamplePath(id)
+
+      // TODO: is there a clean way to not have to read the file so many times?
+      return fs.writeFile(targetPath, data)
+        .then(() => readSample(id))
```

Comment:
wdyt? using `readSample` is clean because then i know it returns the same thing. but this duplicates work, specifically `fs.readFile` twice.

an alternative is to duplicate the audio buffer code here